lib/graphql/list_type.rb

```diff
@@ -17,8 +17,13 @@ def to_s
     "[#{of_type.to_s}]"
   end
 
-  def coerce_input(value)
-    inner_type = of_type
-    value.map { |item| inner_type.coerce_input!(item) }
+  def valid_non_null_input?(value)
+    value = [value] unless value.is_a?(Array)
```

Comment:
Wow, I wonder how I missed that! Kind of a weird behavior IMO.

I suspect that a real query would fail because of this check: https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/static_validation/literal_validator.rb#L6

If you want to add support for that case, could you add an "end-to-end" test for it to make sure it's fully supported? Otherwise, you could remove it from this PR and make it a to-do.